mmseg/datasets/dataset_wrappers.py

```diff
@@ -82,10 +85,20 @@ def evaluate(self, results, logger=None, **kwargs):
                 dataset_idx += 1
                 for k, v in eval_results_per_dataset.items():
                     total_eval_results.update({f'{dataset_idx}_{k}': v})
-
+                    metrics.add(k)
+
+            # calculate the average metric of datasets
+            # if they have same dataset type
+            # e.g. results['mIoU'] = mean(results['0_mIoU], results['1_mIoU]..)
+            if self.have_same_type:
+                for m in metrics:
+                    total_eval_results[m] = np.mean([
+                        total_eval_results[k] for k in total_eval_results
+                        if k.endswith(m)
+                    ])
```

Comment:
I think averaging metrics might be improper, for example, if users would like to split a test dataset into 2 same-type sub-datasets and do evaluation separately, there should be 2 results for sub-datasets and one average result cannot represent model performance. 

I think `save_best` might be turned off when testing datasets separately,

Reply:
Fine, I will revert and use it only in my private branch😂

Reply:
reverted